### ISSUE TITLE
[Question] SaveOnBestTrainingRewardCallback

### ISSUE THREAD

**Issue by danielstankw** [DLR-RM/stable-baselines3#634]
**Important Note: We do not do technical support, nor consulting** and don't answer personal questions per email.
Please post your question on the [RL Discord](https://discord.com/invite/xhfNqQv), [Reddit](https://www.reddit.com/r/reinforcementlearning/) or [Stack Overflow](https://stackoverflow.com/) in that case.

### Question

I want to use the [SaveOnBestTrainingRewardCallback](https://colab.research.google.com/github/Stable-Baselines-Team/rl-colab-notebooks/blob/sb3/monitor_training.ipynb#scrollTo=pUWGZp3i9wyf) given in the stable baselines example but when using SubprocVecEnv with more than 1 env. 
The callback given in the example is not suitable for using with multiple env simultaneously. 
**Did anyone modified it by any chance and would be willing to share a version that works for multiple env?** 

### Additional context

...

### Checklist

- [x] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by danielstankw** [comment]
I want to reopen this issue
`env = SubprocVecEnv([make_robosuite_env(env_id, env_options, i, seed_val) for i in range(num_proc)])
 eval_callback = EvalCallback(env,
                                 best_model_save_path=log_dir,
                                 log_path=log_dir,
                                 eval_freq=3,
                                 deterministic=False,
                                 render=False)
    policy_kwargs = dict(activation_fn=torch.nn.LeakyReLU, net_arch=[32, 32])
    model = PPO('MlpPolicy', env, verbose=1, policy_kwargs=policy_kwargs, n_steps=int(n_steps/num_proc),
                tensorboard_log="./learning_log/ppo_tensorboard/", seed=4)

    model.learn(total_timesteps=10000, tb_log_name="learning", callback=eval_callback, reset_num_timesteps=True)`


I tried the following
and got an error **assert eval_env.num_envs == 1, "You must pass only one environment for evaluation"**

**Comment by danielstankw** [comment]
Ok gotcha,
Thank you very much,
Do you know if the custom callback given in [examples](https://stable-baselines3.readthedocs.io/en/master/guide/examples.html) ` SaveOnBestTrainingRewardCallback` also facilitates multiple envs?

**Comment by araffin** [comment]
> also facilitates multiple envs?

it should work but please don't use it, it is mainly meant as a demo on what you can do with callbacks.
Better is to use `CheckpointCallback` if you cannot evaluate at the same time of training.
I think I will update the doc.

**Comment by danielstankw** [comment]
Thanks, 
I would like to use model that gave me highest reward so why do you say its not good to use the SaveOnBestTrainingRewardCallback? 
CheckpointCallback doesnt give me the functionality I want.

**Comment by araffin** [comment]
> I would like to use model that gave me highest reward 

`SaveOnBestTrainingRewardCallback` only gives you information about a proxy, mean episodic return for the training agent over n training episodes, but the agent is changing between each episode, the true performance at time t can only be known by evaluating it on a separate env for multiple episodes (that's what you can do with `EvalCallback` or do as a post-processing step with `CheckpointCallback`).

If you are doing continuous control, the controller that you are using at the end should be deterministic, which is not the one used for collecting data.

**Comment by danielstankw** [comment]
Thanks a lot for explanation!
